plugin/src/plugin.ts

```diff
@@ -1,37 +1,48 @@
 import type { PluginItem, NodePath } from '@babel/core';
-import { globals } from './commonObjects';
 import type { CallExpression } from '@babel/types';
 import { processForCalleesWorklets } from './processForCalleesWorklets';
 import type { ExplicitWorklet, ReanimatedPluginPass } from './types';
 import { processIfWorkletNode } from './processIfWorkletNode';
 import { processInlineStylesWarning } from './processInlineStylesWarning';
 import { processIfCallback } from './processIfCallback';
+import { addCustomGlobals } from './addCustomGlobals';
+import { initializeGlobals } from './globals';
 
 module.exports = function (): PluginItem {
+  function runWithTaggedExceptions(fun: () => void) {
+    try {
+      fun();
+    } catch (e) {
+      throw new Error('[Reanimated] Babel plugin exception: ' + e);
+    }
+  }
+
   return {
     pre() {
-      // allows adding custom globals such as host-functions
-      if (this.opts != null && Array.isArray(this.opts.globals)) {
-        this.opts.globals.forEach((name: string) => {
-          globals.add(name);
-        });
-      }
+      runWithTaggedExceptions(() => {
+        initializeGlobals();
+        addCustomGlobals.call(this);
```

Comment:
Will it work?
```suggestion
        addCustomGlobals(this);
```

Reply:
No, declaring `this` in a function as an argument in TS is only to assign a type to it, it's not a real argument. The function has to be called with current `this` bound to it and this (imo) is a clean way to do this. Alternate would be

`addCustomGlobals.bind(this)()`

Which just looks bad.